### DISCUSSION TITLE
SemVer 0.3 breaking change

### DISCUSSION THREAD

**Discussion by ShayBox** [thewh1teagle/rookie#14] [Ideas]
I use `rookie` in my library `spotify-lyrics`, and I noticed you made a breaking change in a minor version update in the last two days (I had to manually update the `use` path to the `CookieToString` trait), breaking v0.0.0 of my library from compiling.

[Rust and Cargo use SemVer for version numbers](https://doc.rust-lang.org/cargo/reference/semver.html)

This means prior to v1, your project starts at `0.1.0` and any breaking changes require bumping the middle number, and minor changes that don't break backward compatibility only need to bump the last number.

Switching to v1 is a choice of stability when you think the project won't be making many breaking changes and offsets the numbers to Major.Minor.Patch

EDIT: I wouldn't bother yanking and fixing it, I'm the only person depending on your library right now, but for future reference when your library grows.

**Comment by thewh1teagle** [comment]
I'm sorry about that, and I'm very thankful for your explanation. I wasn't aware of that point regarding changing minor/major versions according to situations like that. It's very important, especially as the project grows. Anyway, it's always in my mind to "try not to make breaking changes," and I'll take that more seriously :)
